src/human_player.c

```diff
@@ -226,17 +193,134 @@ ret_type_e human_process_action_wild_card(void)
     return RET_SUCCESS;
 }
 
+/**
+ * @brief human player discards a action card - Skip
+ *
+ * @param user_input pointer to the user input
+ * @return ret_type_e: RET_SUCCESS on success;
+ *                     RET_INVALID_CARD on invalid input
+ */
+ret_type_e human_process_skip_card(const char* user_input)
+{
+    Card_t card = map_user_input(user_input);
+    if ((card.color == INVALID_COLOR) || (card.name == INVALID_NAME))
+    {
+        return RET_INVALID_INPUT;
+    }
+
+    if (!is_human_card(card))
+    {
+        printf("Entered Card is [%s, %s]! \n", CARD_COLOR_STRING[card.color], CARD_NAME_STRING[card.name]);
+        printf("!!Warning!! Please select a card from your deck! \n");
+        return RET_INVALID_CARD;
+    }
+
+    //If the card is valid, then add the card to the discard_pile
+    remove_card_from_deck(&(g_players[HUMAN].cards_on_hand), card);
+    add_card_at_end(g_discard_pile, card);
+
+    //update g_card_on_table
+    g_card_on_table.color = card.color;
+    g_card_on_table.name = card.name;
+
+    //End turn
+    end_turn(HUMAN);
+    g_card_requested = false;
+
+    //Next turn will be Human turn
+    g_player_on_turn = HUMAN;
+    return RET_SUCCESS;
+}
+
 /**
  * @brief Check if the card discarded by player is an action card
  *
- * @return true on a action card discarding request; otherwise is false.
+ * @param user_input pointer to the user input
+ * @return true on a wild action card discarding request; otherwise is false.
  *                     
  */
-bool is_action_wild_card(const char* user_input)
+bool is_wild_card(const char* user_input)
```

Comment:
Just as suggestion: we can have a function to determine what type of card it is ? Instead of having to check wild/skip separately

Reply:
Good point. Will add it later.